components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/OldTokensCleanDAO.java

```diff
@@ -0,0 +1,130 @@
+            oldAccessTokenObject.setValdityPeriod(resultSet.getLong(12));
+            oldAccessTokenObject.setRefreshTokenValidityPeriod(resultSet.getLong(13));
+            oldAccessTokenObject.setTokenScopeHash(resultSet.getString(14));
+            oldAccessTokenObject.setTokenState(resultSet.getString(15));
+            oldAccessTokenObject.setTokenStateId(resultSet.getString(16));
+            oldAccessTokenObject.setSubjectIdentifier(resultSet.getString(17));
+            oldAccessTokenObject.setAccessTokenHash(resultSet.getString(18));
+            oldAccessTokenObject.setRefreshTokenHash(resultSet.getString(19));
+        }
+        if (OAuthServerConfiguration.getInstance().useRetainOldAccessTokens()) {
+            saveTokenInAuditTable(oldAccessTokenObject, connection);
+        }
+        removeTokenFromMainTable(oldAccessTokenObject.getTokenId(), connection);
+    }
+
+    private void saveTokenInAuditTable(OldAccessTokenDO oldAccessTokenDAO, Connection connection) throws SQLException {
+        PreparedStatement insertintoaudittable = connection.prepareStatement(SQLQueries.STORE_OLD_TOKEN_IN_AUDIT);
+        insertintoaudittable.setString(1, oldAccessTokenDAO.getTokenId());
+        insertintoaudittable.setString(2, oldAccessTokenDAO.getAccessToken());
+        insertintoaudittable.setString(3, oldAccessTokenDAO.getRefreshToken());
+        insertintoaudittable.setInt(4, oldAccessTokenDAO.getConsumerKeyId());
+        insertintoaudittable.setString(5, oldAccessTokenDAO.getAuthzUserValue());
+        insertintoaudittable.setInt(6, oldAccessTokenDAO.getTenantId());
+        insertintoaudittable.setString(7, oldAccessTokenDAO.getUserDomain());
+        insertintoaudittable.setString(8, oldAccessTokenDAO.getUserType());
+        insertintoaudittable.setString(9, oldAccessTokenDAO.getGrantType());
+        insertintoaudittable.setTimestamp(10, oldAccessTokenDAO.getTimeCreated());
+        insertintoaudittable.setTimestamp(11, oldAccessTokenDAO.getRefreshTokenTimeCreated());
+        insertintoaudittable.setLong(12, oldAccessTokenDAO.getValdityPeriod());
+        insertintoaudittable.setLong(13, oldAccessTokenDAO.getRefreshTokenValidityPeriod());
+        insertintoaudittable.setString(14, oldAccessTokenDAO.getTokenScopeHash());
+        insertintoaudittable.setString(15, oldAccessTokenDAO.getTokenState());
+        insertintoaudittable.setString(16, oldAccessTokenDAO.getTokenStateId());
+        insertintoaudittable.setString(17, oldAccessTokenDAO.getSubjectIdentifier());
+        insertintoaudittable.setString(18, oldAccessTokenDAO.getAccessTokenHash());
+        insertintoaudittable.setString(19, oldAccessTokenDAO.getRefreshTokenHash());
+        insertintoaudittable.setTimestamp(20, new Timestamp(System.currentTimeMillis()));
+        insertintoaudittable.execute();
+        if (log.isDebugEnabled()) {
+            log.debug("Successfully saved old access token in audit table. Token ID: " + oldAccessTokenDAO.getTokenId());
```

Comment:
Does this line exceed the max length (120)?

Reply:
fixed in 88fde63